packages/components/atoms/f-button/src/components/Button.vue

```diff
@@ -313,22 +327,63 @@ $btn-sizeXSmall-lineHeight      : 1;
     }
 }
 
+
 /**
- * Modifier – .o-btn--icon
- *
- * Removes unneeded styles from button when only an icon is on it (like a close button)
+ * ==========================================================================
+ * Icon style and modifiers
+ * ==========================================================================
  */
 
 .o-btn--icon {
-    background-color: transparent;
-    border: 0;
-    padding: 0;
-
-    &:hover {
-        background-color: transparent;
+    border-radius: 50%;
+    svg {
+        width: 90%;
+        height: 80%;
     }
-}
 
+ &.o-btn--primary {
+     path {
+          fill: $btn-primary-textColor;
+     }
+ }
+
+ &.o-btn--secondary {
+     path {
+          fill: $btn-secondary-textColor;
+     }
+ }
+
+ &.o-btn--outline {
+     path {
+          fill: $btn-outline-textColor;
+     }
+ }
+
+ &.o-btn--ghost {
+     path {
+          fill: $btn-ghost-textColor;
+     }
+ }
+
+ &.o-btn--link {
+     path {
+          fill: $color-link-default;
+     }
+
+      &:hover {
+          path {
+               fill: $color-link-hover;
+          }
+      }
+
+      &:active,
+      &:focus {
+          path {
+               fill: $color-link-active;
+          }
+      }
+ }
+}
```

Comment:
Just an indentation fix:

```suggestion
    &.o-btn--primary {
        path {
            fill: $btn-primary-textColor;
        }
    }

    &.o-btn--secondary {
        path {
            fill: $btn-secondary-textColor;
        }
    }

    &.o-btn--outline {
        path {
            fill: $btn-outline-textColor;
        }
     }

    &.o-btn--ghost {
        path {
            fill: $btn-ghost-textColor;
        }
    }

    &.o-btn--link {
        path {
            fill: $color-link-default;
        }

        &:hover {
            path {
                fill: $color-link-hover;
            }
        }

        &:active,
        &:focus {
            path {
                fill: $color-link-active;
            }
        }
    }
}
```